worker/manager/src/mirrors_qa_manager/worker.py

```diff
@@ -0,0 +1,309 @@
+
+
+class WorkerManager:
+    """Manager responsible for creating tasks"""
+
+    def __init__(self, worker_id: str) -> None:
+
+        self.worker_id = worker_id
+        self.base_dir = Settings.WORKDIR_FPATH
+
+        # Create the directory where this worker will write their data
+        self.instance_dir = self.base_dir / worker_id
+        self.instance_dir.mkdir(exist_ok=True)
+
+        self.private_key = load_private_key_from_path(Settings.PRIVATE_KEY_FPATH)
+        self.docker = get_docker_client()
+
+        self.host_mounts = query_host_mounts(
+            self.docker,
+            [Settings.WORKDIR_FPATH, Settings.WIREGUARD_KERNEL_MODULES_FPATH],
+        )
+        self.host_workdir = self.host_mounts[Settings.WORKDIR_FPATH]
+
+        self.wg_container_name = f"worker-{self.worker_id}-wireguard"
+        self.task_container_name = f"worker-{self.worker_id}-task"
+        # Create the root directory for binding with the worker's wireguard container
+        self.wg_root_dir = self.instance_dir / "wireguard"
+        self.wg_root_dir.mkdir(exist_ok=True)
+
+        # Create the directory where the configuration file for the interface
+        # will be found.
+        # https://github.com/linuxserver/docker-wireguard/blob/master/root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run#L5
+        self.wg_confs_dir = self.wg_root_dir / "wg_confs"
+        self.wg_confs_dir.mkdir(exist_ok=True)
+        # name of the wireguard interface
+        self.wg_interface = "wg0"
+        self.wg_healthcheck_cmd = [
+            "curl",
+            "--interface",
+            self.wg_interface,
+            "-s",
+            "https://am.i.mullvad.net/json",
+        ]
+        # commands for bringing down/up the interface whenever a new configuration
+        # file is added
+        self.wg_down_cmd = ["wg-quick", "down", self.wg_interface]
+        self.wg_up_cmd = ["wg-quick", "up", self.wg_interface]
+
+        # Check in with the backend API
+        self.auth_credentials = authenticate(self.private_key, self.worker_id)
+
+        # Start the wireguard network container
+        self.wg_container = self.start_wireguard_container(Settings.WIREGUARD_IMAGE)
+
+        # register exit signals
+        self.register_signals()
+
+    def get_host_fpath(self, container_fpath: Path) -> Path:
+        """Determine the host path of a path in the container."""
+        return self.host_workdir / container_fpath.relative_to(Settings.WORKDIR_FPATH)
+
+    def copy_wireguard_conf_file(self, country_code: str | None = None) -> Path:
+        """Copy <country_code>.conf from base directory to instance directory.
+
+        If country_code is None, first file with .conf suffix is copied.
+        Raises:
+            FileNotFoundError: configuration file was not found.
+        """
+        conf_name = None
+
+        if country_code:
+            conf_name = f"{country_code}.conf"
+        else:
+            for fpath in self.base_dir.iterdir():
```

Comment:
```suggestion
            conf_name = next(self.base_dir.rglob("*.conf")).name
```

Reply:
This assumes the configuration file exists and it searches deep into the base directory to look for configuration files. When no file exists, it throws a `StopIteration` error. And also, if the user doesn't place their configuration file in the base dir, it would try to look for configuration files in nested folders (including in its own `instance` folder). 

Also noticed that the wireguard image creates some sample configuration files for a user to modify in a template directory. This could find those files as it is recursing deep.

I think "restricting" the user to keep content only on the root simplifies things.

Reply:
I suppose I could instead use the `glob` method instead and catch the `StopIteration` error